tycho-extras/tycho-pomless/src/main/java/org/eclipse/tycho/pomless/TychoAggregatorMapping.java

```diff
@@ -103,18 +95,13 @@ protected String getPackaging() {
     }
 
     @Override
-    protected void initModel(Model model, Reader artifactReader, File artifactFile)
-            throws ModelParseException, IOException {
+    protected void initModel(Model model, Reader artifactReader, File artifactFile) throws IOException {
         logger.debug("Generate aggregator pom for " + artifactFile);
         try (BufferedReader reader = new BufferedReader(artifactReader)) {
-            String line;
-            while ((line = reader.readLine()) != null) {
-                if (line.startsWith("#") || line.trim().isEmpty()) {
-                    continue;
-                }
+            reader.lines().filter(l -> !l.startsWith("#") && !l.isBlank()).map(String::strip).forEach(line -> {
```

Comment:
You are aware of Oracles warning against side effects in forEach? A more clean functional style would be ...collect(Collectors.toList()) and then modules.addAll(collectedList).

Reply:
@Bananeweizen maybe you can explain what specific problems you see here?

Reply:
I don't see a specific problem in this piece of code, but it's easy to make it problematic. Just have someone extract reader.lines() to a method in the future, and then someone else add a ".parallel()" in that method. Boom! Or someone adds multi-threaded access to the collection that is modified in the forEach. Admitted, that's all far-fetched for this piece of code, but at least I have seen similar chains of refactorings leading to problems due to multiple people changing code independently too often already.
On the other hand, if you use the result.addAll(someStream.collect(toList()) version, it's fault proof for all those mentioned change scenarios and many more.

Reply:
> reader.lines() to a method in the future, and then someone else add a ".parallel()" in that method. Boom!

What will boom? The only thing the javadoc mentions is that order is not guaranteed (something I won't expect anyways) but as module order is not relevant I thing it won't become a problem. That's really a bit hypothetical here.

Reply:
Theoretical you are right, but I think in practice this is not likely a problem. If one chooses to use a parallel stream the entire stream-pipeline should be revised any way. But since that pipeline is not long I think that is something that will be noticed.
However I think it is also not harmful to change it in the suggested way, since creating and iterating over the List will likely not have any performance impact. So I changed it.

Reply:
Maybe then we better stick with a simple iteration instead of "stream" style at all...

Reply:
> Maybe then we better stick with a simple iteration instead of "stream" style at all...

Probably, yes. What do you both think about the current approach?
This does not create a List and also ensures the iteration is done serially even if the stream-pipeline is parallelized.

Reply:
I think creating a list to add it to another list to account for "probably" future refactorings is actually wasted effort, a parallel processing of reading a file which applies trivial operations is even waste of time and most likely "slower", so if streams make the code
a) easier to understand
- or -
b) easier to debug

we should change it if it is just "because we can do it" we should not use it, especially if more complex exception handling is required, streams are an *option* not an requirement nor do they produce "better" or "faster" code.

Reply:
Fully agree.
And I think the latest patch-set reflects that (unfortunately the discussion does not move with subsequent changes.
Just in case you missed that:
```
        try (BufferedReader reader = new BufferedReader(artifactReader)) {
            Stream<String> lines = reader.lines().filter(l -> !l.startsWith("#") && !l.isBlank()).map(String::strip);
            for (String line : (Iterable<String>) lines::iterator) {
                logger.debug("Adding module " + line);
                model.getModules().add(line);
```
Let me know if you think that's not fine.